src/Modules/UngradedModules.jl

```diff
@@ -4543,8 +4590,10 @@ rank   | 0  2  6  6  2
 degree | 4  3  2  1  0
 ```
 """
-function free_resolution(M::SubQuo; ordering::ModuleOrdering = default_ordering(M),
-        length::Int=0, algorithm::Symbol=:fres)
+function free_resolution(M::SubQuo{<:MPolyElem}; 
+    ordering::ModuleOrdering = default_ordering(M),
+    length::Int=0, algorithm::Symbol=:fres
+  )
```

Comment:
@jankoboehm : I restricted this method to the case where it actually works: Polynomials.